### PR TITLE
Docs porting guides 2.0 &  2.3

### PR DESCRIPTION
**Pull request by gundalow**
##### SUMMARY

Improve the user facing documentation to inform people what changes they may need to make when upgrading Ansible.

Currently we only put high level details in CHANGELOG.md, which is fine, as that document is meant to be a summary.

These new pages are designed to have a greater level of detail to help people.

Once we've got a format we like we will create one for `2.4` and keep that update to allow early adopters to follow `devel`

* Create new porting_guide landing page, link from top level index
* Reformat old 2.0 porting guide
* Porting guide for 2.3
* Improve anchors in CHANGELOG to allow easier linking

**Note:** Porting guide for 2.4 has been moved to a separate PR https://github.com/ansible/ansible/pull/24539

##### ISSUE TYPE
 - Docs Pull Request

### CODE REVIEW

**Comment by gundalow** [comment]
ready_for_review

File: CHANGELOG.md [hunk 1]
```diff
@@ -2799,7 +2818,7 @@ the variable is still registered for the host, with the attribute skipped: True.
 * added basename and dirname as Jinja2 filters available to all templates
 * pip works better when sudoing from unprivileged users
 * fix for user creation with groups specification reporting 'changed' incorrectly in some cases
-* fix for some unicode encoding errors in outputting some data in verbose mode
+* fix for some unicode encoding errors in outputing some data in verbose mode
```
Comment (dharmabumstead) [inline CHANGELOG.md]: outputting.

File: CHANGELOG.md [hunk 2]
```diff
@@ -3173,7 +3192,7 @@ New Modules:
 * new LSB facts (release, distro, etc)
 * pause module -- (pause seconds=10) (pause minutes=1) (pause prompt=foo) -- it's an action plugin
 * a module for adding entries to the main crontab (though you may still wish to just drop template files into cron.d)
-* debug module can be used for outputting messages without using 'shell echo'
+* debug module can be used for outputing messages without using 'shell echo'
```
Comment (dharmabumstead) [inline CHANGELOG.md]: outputting

File: docs/docsite/rst/porting_guide_2.3.rst [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+           - command: /bin/false
+           - debug: msg='I also never execute :-('
+         always:
+           - debug: msg="this always executes"
+
+
+Use of multiple tags
+--------------------
+
+Specifying ``--tags`` (or ``--skip-tags``) multiple times on the command line currently leads to the last one overriding all the previous ones. This behaviour is deprecated. In the future, if you specify --tags multiple times the tags will be merged together. From now on, using ``--tags`` multiple times on one command line will emit a deprecation warning. Setting the ``merge_multiple_cli_tags`` option to True in the ``ansible.cfg`` file will enable the new behaviour.
```
Comment (dharmabumstead) [inline docs/docsite/rst/porting_guide_2.3.rst]: change "the last one" to "the last specified tag" and "all the previous ones" to "all the other specified tags".

**Review by dharmabumstead — CHANGES_REQUESTED** [review]
OK with a few requested edits.

**Comment by gundalow** [comment]
Cherry picked into `https://github.com/ansible/ansible/pull/27632` in 
https://github.com/ansible/ansible/pull/27632

**Pull request merged**